File: ice0208/array_string/334.py
class Solution:
    def increasingTriplet(self, nums: list[int]) -> bool:
        a, b = 2**31 - 1, 2**31 - 1

        for num in nums:
            if num <= a:
                a = num
                continue
            elif num <= b:
                b = num
                continue
            else:
                return True

        return False


if __name__ == "__main__":
    print(Solution().increasingTriplet([1, 2, 3, 4, 5]))
    print(Solution().increasingTriplet([5, 4, 3, 2, 1]))
    print(Solution().increasingTriplet([2, 1, 5, 0, 4, 6]))
    print(Solution().increasingTriplet([20, 100, 10, 12, 5, 13]))
    print(Solution().increasingTriplet([0, 4, 2, 1, 0, -1, -3]))

File: moonpoison/array_string/334.cpp
#include<bits/stdc++.h>
using namespace std;
class Solution {
public:
    bool increasingTriplet(vector<int>& nums) 
    {
        cin.tie(0); cout.tie(0);
        ios::sync_with_stdio(false);
        int first=INT_MAX;
        int second=INT_MAX;
        for(int i=0;i<nums.size();++i)
        {
            if(nums[i]<=first) first=nums[i];
            else if(nums[i]<=second) second=nums[i];
            else return true;
        }
        return false;
    }
};
